### PR TITLE
Refactoring based on sonarcloud quality

### PR DESCRIPTION
**Pull request by MartinSchmidt**
General cleanup based on sonarcloud recommendations

### CODE REVIEW

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs [hunk 1]
```diff
@@ -208,6 +208,47 @@ public async Task GetAllOwnedCertificates_AllSliceStates(SliceState sliceState,
...
+    {
+        await _connection.ExecuteAsync("DELETE FROM ReceivedSlices");
+
+        var slice = await _repository.GetTop1ReceivedSlice();
+
+        slice.Should().BeNull();
+    }
+
+    [Fact]
+    public async Task GetTop1ReceivedSlice_ReturnsSlice_AndRemoveSlice()
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs:223]: Why did you add this test now?

File: src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs [hunk 2]
```diff
@@ -133,7 +126,7 @@ private async Task<IEnumerable<Slice>> ReserveRequiredSlices(IEnumerable<Slice>
         }
         _unitOfWork.Commit();
 
-        _logger.LogTrace($"{takenSlices.Count} slices reserved.");
+        _logger.LogTrace("{count} slices reserved.", takenSlices.Count);
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs:129]: Explain this difference again :)

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs [hunk 3]
```diff
@@ -11,47 +11,104 @@ namespace ProjectOrigin.WalletSystem.Server.Repositories;
...
 {
-    private IDbConnection _connection;
+    private readonly IDbConnection _connection;
 
     public CertificateRepository(IDbConnection connection) => this._connection = connection;
 
     public async Task InsertSlice(Slice newSlice)
     {
-        var registryId = await _connection.QuerySingleAsync<Guid>("SELECT Id FROM Registries WHERE Name = @registry", new { newSlice.Registry });
+        var registryId = await _connection.QuerySingleAsync<Guid>(
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:20]: Did sonar complain about this?

File: src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs (hunk 2 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs:129]: There is also an example here [example](https://serilog.net)

old version AND new version in text format output:

```log
08:55:22 [Trace] 1 slices reserved.
```

but in json output the format would be similar to 

## OLD
```json
{
    "Timestamp": "2023-09-28T06:55:22.9438887+00:00",
    "Level": "Trace",
    "MessageTemplate": "1 slices reserved.",
    "Properties": {
        ...
    }
}
```

## New 
```json
{
    "Timestamp": "2023-09-28T06:55:22.9438887+00:00",
    "Level": "Trace",
    "MessageTemplate": "{count} slices reserved.",
    "Properties": {
        "count": "1",
        ...
    }
}
```

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs (hunk 3 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:20]: Nope, just prefer the formatting.

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs (hunk 1 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs:223]: The method was untested after some previous changes.

**Review by tnickelsen — APPROVED** [review]
